erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -788,124 +562,20 @@ WebRTCEvent WebRtcConnection::getCurrentState() {
 }
 
 void WebRtcConnection::getJSONStats(std::function<void(std::string)> callback) {
-  asyncTask([callback] (std::shared_ptr<WebRtcConnection> connection) {
-    std::string requested_stats = connection->stats_->getStats();
-    //  ELOG_DEBUG("%s message: Stats, stats: %s", connection->toLog(), requested_stats.c_str());
-    callback(requested_stats);
-  });
-}
-
-void WebRtcConnection::changeDeliverPayloadType(DataPacket *dp, packetType type) {
-  RtpHeader* h = reinterpret_cast<RtpHeader*>(dp->data);
-  RtcpHeader *chead = reinterpret_cast<RtcpHeader*>(dp->data);
-  if (!chead->isRtcp()) {
-      int internalPT = h->getPayloadType();
-      int externalPT = internalPT;
-      if (type == AUDIO_PACKET) {
-          externalPT = remoteSdp_.getAudioExternalPT(internalPT);
-      } else if (type == VIDEO_PACKET) {
-          externalPT = remoteSdp_.getVideoExternalPT(externalPT);
-      }
-      if (internalPT != externalPT) {
-          h->setPayloadType(externalPT);
-      }
-  }
-}
-
-// parses incoming payload type, replaces occurence in buf
-void WebRtcConnection::parseIncomingPayloadType(char *buf, int len, packetType type) {
-  RtcpHeader* chead = reinterpret_cast<RtcpHeader*>(buf);
-  RtpHeader* h = reinterpret_cast<RtpHeader*>(buf);
-  if (!chead->isRtcp()) {
-    int externalPT = h->getPayloadType();
-    int internalPT = externalPT;
-    if (type == AUDIO_PACKET) {
-      internalPT = remoteSdp_.getAudioInternalPT(externalPT);
-    } else if (type == VIDEO_PACKET) {
-      internalPT = remoteSdp_.getVideoInternalPT(externalPT);
-    }
-    if (externalPT != internalPT) {
-      h->setPayloadType(internalPT);
-    } else {
-//        ELOG_WARN("onTransportData did not find mapping for %i", externalPT);
-    }
-  }
+  media_stream_->getJSONStats(callback);
 }
 
 void WebRtcConnection::write(std::shared_ptr<DataPacket> packet) {
-  Transport *transport = (bundle_ || packet->type == VIDEO_PACKET) ? videoTransport_.get() :
-                                                                     audioTransport_.get();
+  Transport *transport = (bundle_ || packet->type == VIDEO_PACKET) ? videoTransport_.get() : audioTransport_.get();
```

Comment:
in a future, with multiple stream, we would need to use a worker and an async task here to protect memory, am I right?